lib/action_client/base.rb

```diff
@@ -3,34 +3,61 @@ class Base < AbstractController::Base
     include AbstractController::Rendering
     include ActionView::Layouts
 
-    cattr_accessor :defaults,
+    class_attribute :defaults,
       instance_accessor: true,
       default: ActiveSupport::OrderedOptions.new
 
+    class_attribute :middleware,
+      instance_accessor: true
+
+    class_attribute :after_actions,
+      instance_accessor: true,
+      default: []
+
     class << self
+      def inherited(descendant)
+        descendant.middleware = Rails.configuration.action_client.middleware.dup
+        descendant.after_actions = after_actions.dup
+      end
+
       def default(options)
         options.each do |key, value|
           defaults[key] = value
         end
       end
 
+      def after_submit(with_status: nil, &block)
+        if with_status
+          after_submit do |status, headers, body|
+            rescued_codes = Array(with_status).map do |status_code|
+              Rack::Utils.status_code(status_code)
+            end
+
+            if rescued_codes.any? { |code| Array(code).include?(status) }
```

Comment:
This block, is itching for an extraction.

Reply:
@georgebrock I've extracted `AfterSubmit`: https://github.com/thoughtbot/action_client/pull/3/files#diff-7163e32d86fe0c8e00c3d3e7bd878a1e

I'm not confident that it's the right extraction, though.

Do you have any ideas?